src/Plugin/Block/SamlLogoutBlock.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\stanford_samlauth\Plugin\Block;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Provides a 'Saml Logout Block' block.
+ *
+ * @Block(
+ *  id = "stanford_samlauth_logout_block",
+ *  admin_label = @Translation("SAML SUNetID Logout Block")
+ * )
+ */
+class SamlLogoutBlock extends BlockBase implements ContainerFactoryPluginInterface {
```

Comment:
Needs tests. See the login block tests: https://github.com/SU-SWS/stanford_samlauth/blob/1.x/tests/src/Unit/Plugin/Block/SamlLoginBlockTest.php